sources/Interop/D3D12MemoryAllocator/src/Allocation.cs

```diff
@@ -0,0 +1,455 @@
+        /// <returns>The name associated with the allocation object.</returns>
+        [return: NativeTypeName("LPCWSTR")]
+        public readonly char* GetName() { return m_Name; }
+
+        /// <summary>
+        /// Returns `TRUE` if the memory of the allocation was filled with zeros when the allocation was created.
+        /// <para>Returns `TRUE` only if the allocator is sure that the entire memory where the allocation was created was filled with zeros at the moment the allocation was made.</para>
+        /// <para>
+        /// Returns `FALSE` if the memory could potentially contain garbage data.
+        /// If it's a render-target or depth-stencil texture, it then needs proper initialization with `ClearRenderTargetView`, `ClearDepthStencilView`, `DiscardResource`,
+        /// or a copy operation, as described on page: [ID3D12Device::CreatePlacedResource method - Notes on the required resource initialization] (https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-createplacedresource#notes-on-the-required-resource-initialization).
+        /// Please note that rendering a fullscreen triangle or quad to the texture as a render target is not a proper way of initialization!
+        /// </para>
+        /// <para>
+        /// See also articles:
+        /// ["Coming to DirectX 12: More control over memory allocation"] (https://devblogs.microsoft.com/directx/coming-to-directx-12-more-control-over-memory-allocation/),
+        /// ["Initializing DX12 Textures After Allocation and Aliasing"] (https://asawicki.info/news_1724_initializing_dx12_textures_after_allocation_and_aliasing).
+        /// </para>
+        /// </summary>
+        /// <returns>Whether the memory of the allocation was filled with zeros.</returns>
+        [return: NativeTypeName("BOOL")]
+        public readonly int WasZeroInitialized() { return m_PackedData.WasZeroInitialized(); }
+
+        internal enum Type
+        {
+            TYPE_COMMITTED,
+            TYPE_PLACED,
+            TYPE_HEAP,
+            TYPE_COUNT
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        internal struct _Anonymous_e__Union
+        {
+            [FieldOffset(0)] public m_Committed_t m_Committed;
+            [FieldOffset(0)] public m_Placed_t m_Placed;
+            [FieldOffset(0)] public m_Heap_t m_Heap;
+
+            public struct m_Committed_t
+            {
+                public D3D12_HEAP_TYPE heapType;
+            }
+
+            public struct m_Placed_t
+            {
+                [NativeTypeName("UINT64")] public ulong offset;
+                public NormalBlock* block;
+            }
+
+            public struct m_Heap_t
+            {
+                public D3D12_HEAP_TYPE heapType;
+                public ID3D12Heap* heap;
+            }
+        }
+
+        internal partial struct PackedData
+        {
+            ulong __value;
+
+            public readonly new Type GetType() { return (Type)m_Type; }
```

Comment:
nit: braces must be on their own line, outside of lambdas or collection initializers

For these methods, use lambda expressions
```suggestion
            public readonly new Type GetType() => (Type)m_Type;
```

Reply:
Done in one of the last commits, forgot which one in particular.